File: EHRAVersion2/Backend/src/main/java/com/ehrassist/service/PatientService.java
package com.ehrassist.service;

import org.hl7.fhir.r4.model.Bundle;
import org.hl7.fhir.r4.model.Patient;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface PatientService {
    Patient getById(UUID id);
    Bundle search(UUID id, String family, String given, String gender, String birthdate, String email, Pageable pageable);
    Patient create(Patient resource);
    Patient update(UUID id, Patient resource);
    void delete(UUID id);
}

File: EHRAVersion2/ehrassist-patient-portal/src/services/patientService.js
import apiClient from '../config/api';

export const patientService = {
  getById: async (id) => {
    const response = await apiClient.get(`/Patient/${id}`);
    return response.data;
  },

  search: async (params) => {
    const response = await apiClient.get('/Patient', { params });
    return response.data;
  },

  update: async (id, patientData) => {
    const response = await apiClient.put(`/Patient/${id}`, JSON.stringify(patientData));
    return response.data;
  },

  create: async (patientData) => {
    const response = await apiClient.post('/Patient', JSON.stringify(patientData));
    return response.data;
  },

  delete: async (id) => {
    await apiClient.delete(`/Patient/${id}`);
  }
};
